modules/iam-identity/src/test/java/com/ibm/cloud/platform_services/iam_identity/v1/IamIdentityIT.java

```diff
@@ -661,12 +678,565 @@ public void testGetServiceIdNegative() throws Exception {
+            do {
+                // Get the "next" page of results.
+                ListProfileOptions listProfileOptions = new ListProfileOptions.Builder()
+                        .accountId(ACCOUNT_ID)
+                        .pagetoken(pagetoken)
+                        .pagesize(pagesize)
+                        .includeHistory(false)
+                        .build();
+
+                Response<TrustedProfilesList> response = service.listProfile(listProfileOptions).execute();
+                assertNotNull(response);
+                assertEquals(response.getStatusCode(), 200);
+
+                TrustedProfilesList profilesListResult = response.getResult();
+                assertNotNull(profilesListResult);
+
+                assertEquals(profilesListResult.getLimit(), Long.valueOf(pagesize));
+
+                // Walk through the returned results
+                for (TrustedProfile profile : profilesListResult.getProfiles()) {
+                    if (PROFILE_NAME_1.equals(profile.getName()) || PROFILE_NAME_2.equals(profile.getName())) {
+                        profiles.add(profile);
+                    }
+                }
+
+                // Retrieve the pagetoken value for the next page of results.
+                pagetoken = getPageTokenFromURL(profilesListResult.getNext());
+            } while (pagetoken != null);
+
+            assertEquals(profiles.size(), 2);
+        } catch (ServiceResponseException e) {
+            fail(String.format("Service returned status code %d: %s\nError details: %s",
+                    e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+    @Test(dependsOnMethods = { "testListProfiles" })
+    public void testUpdateProfile() throws Exception {
+        assertNotNull(profileId1);
+        assertNotNull(profileEtag);
+        try {
+            String newDescription = "updated description";
+            UpdateProfileOptions updateProfileOptions = new UpdateProfileOptions.Builder()
+                    .profileId(profileId1)
+                    .ifMatch(profileEtag)
+                    .description(newDescription)
+                    .build();
+
+            Response<TrustedProfile> response = service.updateProfile(updateProfileOptions).execute();
+            assertNotNull(response);
+            assertEquals(response.getStatusCode(), 200);
+
+            TrustedProfile profileResult = response.getResult();
+            assertNotNull(profileResult);
+
+            assertEquals(profileResult.getDescription(), newDescription);
+            assertEquals(profileResult.getName(), PROFILE_NAME_1);
+        } catch (ServiceResponseException e) {
+            fail(String.format("Service returned status code %d: %s\nError details: %s",
+                    e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+        }
+    }
+
+    @Test(dependsOnMethods = { "testUpdateProfile" })
+    public void testDeleteProfile1() throws Exception {
+        sleep(1);
```

Comment:
Why are we waiting here? Possibly I missed something.

Reply:
this way we make sure the delete action happens at last.. we handle the apikeys the same way.